rotkehlchen/db/custom_assets.py

```diff
@@ -0,0 +1,111 @@
+            return [CustomAsset.deserialize_from_db(entry) for entry in cursor]
+
+    def get_custom_assets_and_limit_info(
+            self,
+            filter_query: CustomAssetsFilterQuery,
+    ) -> Tuple[List[CustomAsset], int, int]:
+        """
+        Returns a list of `CustomAsset`, a count of the assets that match
+        the query and the count of custom assets in the DB.
+        """
+        entries = self._get_custom_assets(filter_query=filter_query)
+        with GlobalDBHandler().conn.read_ctx() as cursor:
+            query, bindings = filter_query.prepare(with_pagination=False)
+            query = 'SELECT COUNT(*) from custom_assets AS A JOIN assets AS B ON A.identifier = B.identifier ' + query  # noqa: E501
+            entries_found = cursor.execute(query, bindings).fetchone()[0]
+            entries_total = cursor.execute('SELECT COUNT(*) FROM custom_assets;').fetchone()[0]
+            return entries, entries_found, entries_total
+
+    def add_custom_asset(self, custom_asset: CustomAsset) -> str:
+        """
+        Adds a custom new custom asset into `assets` and `custom_assets` table.
+        May raise:
+        - InputError if the combination of the name & type / asset with identifier already exist.
+        """
+        self.check_if_custom_asset_exists(custom_asset)
+        with GlobalDBHandler().conn.write_ctx() as global_db_write_cursor, self.db.user_write() as db_write_cursor:  # noqa: E501
+            global_db_write_cursor.execute(
+                'INSERT INTO assets(identifier, name, type) VALUES (?, ?, ?)',  # noqa: E501
+                (
+                    custom_asset.identifier,
+                    custom_asset.name,
+                    AssetType.CUSTOM_ASSET.serialize_for_db(),
+                ),
+            )
+            global_db_write_cursor.execute(
+                'INSERT INTO custom_assets(identifier, type, notes) VALUES(?, ?, ?)',
+                custom_asset.serialize_for_db(),
+            )
+            self.db.add_asset_identifiers(db_write_cursor, [custom_asset.identifier])
+        return custom_asset.identifier
+
+    def edit_custom_asset(self, custom_asset: CustomAsset) -> None:
+        """
+        Edits a custom asset.
+        May raise:
+        - InputError if editing a custom asset that does not exist or combination
+        of name and type is already present.
+        """
+        self.check_if_custom_asset_exists(custom_asset)
+        with GlobalDBHandler().conn.write_ctx() as write_cursor:
+            write_cursor.execute(
+                'UPDATE assets SET name=? WHERE identifier=?',
+                (custom_asset.name, custom_asset.identifier),
+            )
+            write_cursor.execute(
+                'UPDATE custom_assets SET notes=?, type=? WHERE identifier=?',
+                (custom_asset.notes, custom_asset.custom_asset_type, custom_asset.identifier),
+            )
+            if write_cursor.rowcount == 0:
```

Comment:
can this happen? Don't you already check for this with `.check_if_custom_asset_exists()`? Why do both?

Reply:
`.check_if_custom_asset_exists()` only checks if the combination of name & asset exists. It doesn't check for the identifier.

Reply:
Why does it not also check for identifier? Didn't you just add the identifier check right now?

Reply:
`_raise_if_custom_asset_exists()` simply checks if the combination of a name and type for a custom asset exist.
If it does, check if the edited asset is the same as the match that exists, don't raise an error. Otherwise, raise an error

Reply:
Okay I see. Then can you add a comment under or above the if explaining this. Explaining in short why you need this check while you already do a `raise_if_custom_asset_exists` above?